src/distributions/weighted_index.rs

```diff
@@ -102,13 +102,13 @@ impl<X: SampleUniform + PartialOrd> WeightedIndex<X> {
         let mut total_weight: X = iter.next().ok_or(WeightedError::NoItem)?.borrow().clone();
 
         let zero = <X as Default>::default();
-        if total_weight < zero {
+        if matches!(total_weight.partial_cmp(&zero), Some(Ordering::Less) | None) {
             return Err(WeightedError::InvalidWeight);
         }
```

Comment:
```suggestion
        if let Some(Ordering::Less) | None = total_weight.partial_cmp(&zero) {
            return Err(WeightedError::InvalidWeight);
        }
```

Reply:
Oh wow, didn't even know that was possible.